test/ComponentInstallerTest.php

```diff
@@ -85,20 +81,18 @@ protected function setUp(): void
             vfsStream::url('project')
         );
 
-        /** @psalm-var Composer&MockObject $composer */
         $composer       = $this->createMock(Composer::class);
         $this->composer = $composer;
 
-        /** @psalm-var RootPackageInterface&MockObject $rootPackage */
-        $rootPackage       = $this->createMock(RootPackageInterface::class);
+        $rootPackage       = $this->createMock(RootPackage::class);
```

Comment:
We have to mock the `RootPackage` directly due to https://github.com/composer/composer/commit/e3d99cac5928d72859fac8300255c9566a26e9d5#diff-b6a84ccae2ddc637751457ddb13a3411178a3ebbfc147cc3857b3a4dc57e1fbfR217-R219